File: Python/482.py
"""
@no 482
@name License Key Formatting
"""
class Solution:
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        string = S.replace('-', '')
        tmp = ''

        count = 0
        for i in range(len(string) - 1, -1, -1):
            tmp += string[i].upper()
            count += 1
            if count == K and i != 0:
                tmp += '-'
                count = 0

        return tmp[::-1]

File: Java/482.java
/**
 * @no 482
 * @name License Key Formatting
 */
class Solution {
    public static String licenseKeyFormatting(String s, int k) {
        String source = new StringBuffer(s.replace("-", "").toUpperCase()).reverse().toString();
        StringBuilder resultBuilder = new StringBuilder();
        for (int i = 0; i < source.length(); i++) {
            if (i % k == 0 && i != 0) {
                resultBuilder.append('-');
            }
            resultBuilder.append(source.charAt(i));
        }
        return new StringBuffer(resultBuilder.toString()).reverse().toString();
    }
}